File: src/JavaClient/ClientExecutable.java
import WordUnscramblerApp.*;
import model.User;
import org.omg.CosNaming.*;
import org.omg.CosNaming.NamingContextPackage.*;
import org.omg.CORBA.*;
import java.util.Scanner;

import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;

public class ClientExecutable extends Application {
    public static WordUnscrambler impl;
    public static model.User currUser;

    @Override
    public void start(Stage primaryStage) throws Exception {
        Parent root = FXMLLoader.load(getClass().getClassLoader().getResource("resources/view/Menu.fxml"));
        Scene scene = new Scene(root);
        primaryStage.setTitle("Textra");
        primaryStage.setScene(scene);
        primaryStage.show();
        primaryStage.setResizable(false);
    }

    public static void main(String[] args) {
        try {
            // create and initialize the ORB
            ORB orb = ORB.init(args, null);

            // get the root naming context
            org.omg.CORBA.Object objRef = orb.resolve_initial_references("NameService");

            // Use NamingContextExt instead of NamingContext. This is part
            // of the Interoperable naming Service.
            NamingContextExt ncRef = NamingContextExtHelper.narrow(objRef);

            // resolve the Object Reference in Naming
            String name = "Hello";
            impl = WordUnscramblerHelper.narrow(ncRef.resolve_str(name));

            System.out.println("Client successfuly connected");
            User.impl = impl;

            Scanner in = new Scanner(System.in);
            in.nextLine();

            currUser = null;
            launch(args);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: src/PythonClient/ClientExecutable.py
import CORBA, WordUnscramblerApp, CosNaming

def main(args):
    orb = CORBA.ORB_init(args, CORBA.ORB_ID)
    print("----------------------")
    print("CLIENT STARTING...")
    print("----------------------")

    obj = orb.resolve_initial_references("NameService")
    rootContext = obj._narrow(CosNaming.NamingContext)

    if rootContext is None:
        print ("Failed to narrow the root naming context")
        sys.exit(1)

    name = [CosNaming.NameComponent("Hello", ""), ]
    try:
        obj = rootContext.resolve(name)
    except CosNaming.NamingContext.NotFound as ex:
        print ("Name not found")
        sys.exit(1)

    eo = obj._narrow(WordUnscramblerApp.WordUnscrambler)

    if eo is None:
        print ("Object reference is not an WordUnscramblerApp::WordUnscrambler")
        sys.exit(1)

    print(eo.logIn("hotdog"))

    print("----------------------")
    print("ORB DESTROYED")
    orb.destroy()


if __name__ == "__main__":
    import sys
    main(sys.argv)
    print("----------------------")
    print("CLIENT NOW EXITING...")
    print("----------------------")